relay-server/src/actors/project_cache.rs

```diff
@@ -625,36 +623,55 @@ impl ProjectCacheBroker {
     }
 
     /// Handles the processing of the provided envelope.
-    fn handle_processing(
-        &mut self,
-        state: Arc<ProjectState>,
-        sampling_state: Option<Arc<ProjectState>>,
-        envelope: Box<Envelope>,
-        envelope_context: EnvelopeContext,
-    ) {
+    ///
+    /// The following pre-conditions must be met before calling this function:
+    /// - Envelope's project state must be cached and valid.
+    /// - Optional: if dynamic sampling keys exists, the dyanmic project state must be cached and valid.
```

Comment:
```suggestion
    /// - If dynamic sampling keys exists, the sampling project state must be cached and valid.
```

Reply:
also done in [56f32b8](https://github.com/getsentry/relay/pull/1907/commits/56f32b8f089d0932fb8aa463814474fb79edac27)